lang/syn/src/idl/file.rs

```diff
@@ -470,11 +463,39 @@ fn parse_ty_defs(ctx: &CrateContext) -> Result<Vec<IdlTypeDefinition>> {
 }
 
 // Replace variable array lengths with values
-fn resolve_variable_array_length(ctx: &CrateContext, tts_string: String) -> String {
-    for constant in ctx.consts() {
-        if constant.ty.to_token_stream().to_string() == "usize"
-            && tts_string.contains(&constant.ident.to_string())
-        {
+fn resolve_variable_array_lengths(ctx: &CrateContext, mut tts_string: String) -> String {
+    for constant in ctx.consts().filter(|c| match *c.ty {
+        // Filter to only those consts that are of type usize or could be cast to usize
+        syn::Type::Path(ref p) => {
+            let segment = p.path.segments.last().unwrap();
+            matches!(
+                segment.ident.to_string().as_str(),
+                "usize"
+                    | "u8"
+                    | "u16"
+                    | "u32"
+                    | "u64"
+                    | "u128"
+                    | "isize"
+                    | "i8"
+                    | "i16"
+                    | "i32"
+                    | "i64"
+                    | "i128"
+            )
+        }
+        _ => false,
+    }) {
+        let size_string = format!("{}]", &constant.ident.to_string());
+        let cast_size_string = format!("{} as usize]", &constant.ident.to_string());
```

Comment:
wdyt about removing the whitespace here and in `tts_string` and then comparing those instead? would really suck to debug an error that's caused by an additional whitespace between `as` and `usize`

Reply:
No sympathy for people who don't use code formatters. 😈 

I think it's an unlikely case but I'll add it if you like. Regex would probably have fewer edge cases but adding that seems a bit over the top.

Reply:
No sympathy for people who don't use code formatters. 😈 

I think it's an unlikely case but I'll add it if you like. Regex would probably have fewer edge cases but adding that seems a bit over the top.

Reply:
some people run their formatters before they commit and not during coding. still unlikely that it even happens but doesnt hurt to add it

agree that regex seems over the top